### PR TITLE
add retry for certain transaction errors

### PR DESCRIPTION
**Pull request by tolbrino**

### CODE REVIEW

File: api/pages/api/faucet/[environment]/[address]/native.ts [hunk 1]
```diff
@@ -44,24 +47,20 @@ export default async function handler(
...
-        .json({
-          err: `Environment ${actualEnvironment} has an invalid configurated network (${network}), please ensure the provided network by the environment exists`,
-        });
+      return res.status(501).json({
+        err: `Environment ${actualEnvironment} has an invalid configurated network (${network}), please ensure the provided network by the environment exists`,
+      });
 
     const networkConfig = protocolConfig.networks[network];
     const provider = new providers.JsonRpcProvider(
-      "https://goerli.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161"
```
Comment (robertkiel) [inline api/pages/api/faucet/[environment]/[address]/native.ts:64]: lol ...

File: api/utils/wallet.ts [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
+import { providers, errors, BigNumber } from "ethers";
 import Redis from "ioredis";
 
+import type { Signer } from "@ethersproject/abstract-signer";
+import type {
+  TransactionRequest,
+  TransactionReceipt,
+} from "@ethersproject/abstract-provider";
+
+export async function performTransaction(
```
Comment (robertkiel) [inline api/utils/wallet.ts:10]: Don't think it's a good idea to implement this function recursively, especially as it uses dangerous libraries like `ethers` for RPC calls

File: api/utils/wallet.ts [hunk 3]
```diff
@@ -11,11 +49,19 @@ export const getLockedTransaction = async (
...
   const storedNonce = await client.get(key);
+  const chainNonce = await provider.getTransactionCount(address, "latest");
   let nonce;
   if (storedNonce === null) {
-    nonce = await provider.getTransactionCount(address, "latest");
+    // initialize nonce once from the chain
+    nonce = chainNonce;
   } else {
+    // ensure our stored nonce is up-to-date, if not overwrite it
     nonce = +storedNonce;
```
Comment (robertkiel) [inline api/utils/wallet.ts:59]: Isn't the `+` operator a no-op here?

Comment (tolbrino) [reply api/utils/wallet.ts:59]: at that point the value is a string, we need a number

**Pull request merged**